src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -1883,6 +1885,51 @@ SequenceDependencyCommandList(Oid relationId)
 }
 
 
+/*
+ * IdentitySequenceDependencyCommandList generate a command to execute a UDF (WORKER_MODIFY_IDENTITY_COLUMNS) on workers
+ * to modify the identity columns min/max values to produce unique values on workers.
+ */
+List *
+IdentitySequenceDependencyCommandList(Oid targetRelationId)
+{
+	List *commandList = NIL;
+
+	Relation relation = relation_open(targetRelationId, AccessShareLock);
```

Comment:
Hmm, I find this pattern of closing the relation immediately a little unusual. I see that we have it in other identity column related functions as well.

Are you sure that PG does not free tupleDescriptor when relation_close happens?